Source/WebCore/html/HTMLFieldSetElement.cpp

```diff
@@ -63,18 +63,22 @@ Ref<HTMLFieldSetElement> HTMLFieldSetElement::create(const QualifiedName& tagNam
 
 static void updateFromControlElementsAncestorDisabledStateUnder(HTMLElement& startNode, bool isDisabled)
 {
-    RefPtr<HTMLFormControlElement> control;
-    if (is<HTMLFormControlElement>(startNode))
-        control = &downcast<HTMLFormControlElement>(startNode);
-    else
-        control = Traversal<HTMLFormControlElement>::firstWithin(startNode);
-    while (control) {
-        control->setAncestorDisabled(isDisabled);
-        // Don't call setAncestorDisabled(false) on form controls inside disabled fieldsets.
-        if (is<HTMLFieldSetElement>(*control) && control->hasAttributeWithoutSynchronization(disabledAttr))
-            control = Traversal<HTMLFormControlElement>::nextSkippingChildren(*control, &startNode);
+    auto range = inclusiveDescendantsOfType<Element>(startNode);
+    auto it = range.begin();
+
+    // This preserves status-quo established before https://github.com/WebKit/WebKit/pull/4988:
+    // setAncestorDisabled() may modify shadow DOM of <input> / <textarea> elements, but we don't care.
+    it.dropAssertions();
```

Comment:
Why do we need to drop assertions? That seems bad.

Reply:
Only because `<input>`'s / `<textarea>`'s shadow DOM could be modified by `setAncestorDisabled()`. Nothing that is observable by a developer. Before this change we didn't have assertions as well since we didn't use the `ElementIterator`, which is super neat.

Reply:
Ah, okay. Sounds good to me.